script/tasks/diamondSyncWhitelist.sh

```diff
@@ -567,112 +588,116 @@ function diamondSyncWhitelist {
+                echo "  [$i]: ${UPDATED_PAIRS[$i]}"
+              fi
+            done
+          fi
+
           if [[ $verify_exit_code -ne 0 ]]; then
             printf '\033[0;33m%s\033[0m\n' "⚠️  [$NETWORK] Could not verify whitelist state (getAllContractSelectorPairs failed)"
             printf '\033[0;36m%s\033[0m\n' "💡 Transaction succeeded but verification skipped - pairs should be whitelisted"
             return 0
           fi
-          
+
           # Verify pairs were added correctly
           # NEW_PAIRS format: array of "address|selector" strings
           # UPDATED_PAIRS format: array of "address|selector" strings (from getCurrentWhitelistedPairs)
           # Both can be in different orders, so we need exact matching
-          
+
           local VERIFIED_COUNT=0
           local MISSING_COUNT=0
-          
-          echo ""
-          echo "=== VERIFICATION DEBUG ==="
-          echo "Comparing ${#NEW_PAIRS[@]} NEW_PAIRS against ${#UPDATED_PAIRS[@]} UPDATED_PAIRS"
-          echo ""
-          
-          # Normalize both arrays to lowercase for comparison
-          local NORMALIZED_NEW=()
-          for pair in "${NEW_PAIRS[@]}"; do
-            NORMALIZED_NEW+=("$(echo "$pair" | tr '[:upper:]' '[:lower:]')")
-          done
-          
-          local NORMALIZED_UPDATED=()
-          for pair in "${UPDATED_PAIRS[@]}"; do
-            NORMALIZED_UPDATED+=("$(echo "$pair" | tr '[:upper:]' '[:lower:]')")
-          done
-          
-          # Show sample of what we're comparing (first 3 from each array)
-          echo "Sample NEW_PAIRS (first 3):"
-          for i in 0 1 2; do
-            if [[ $i -lt ${#NORMALIZED_NEW[@]} ]]; then
-              echo "  NEW[$i]: ${NORMALIZED_NEW[$i]}"
-            fi
-          done
-          echo ""
-          echo "Sample UPDATED_PAIRS (first 3):"
-          for i in 0 1 2; do
-            if [[ $i -lt ${#NORMALIZED_UPDATED[@]} ]]; then
-              echo "  UPDATED[$i]: ${NORMALIZED_UPDATED[$i]}"
-            fi
-          done
-          echo ""
-          
-          # Check each NEW pair against UPDATED pairs
-          local checked_count=0
-          for new_pair in "${NORMALIZED_NEW[@]}"; do
-            local FOUND=false
-            
-            # Show detailed comparison for first 3 pairs
-            if [[ $checked_count -lt 3 ]]; then
-              echo "Checking NEW pair [$checked_count]: '$new_pair'"
-              echo "  Searching in ${#NORMALIZED_UPDATED[@]} UPDATED pairs..."
-            fi
-            
-              # Search for exact match in UPDATED_PAIRS
-              for updated_pair in "${NORMALIZED_UPDATED[@]}"; do
-                if [[ "$new_pair" == "$updated_pair" ]]; then
-                FOUND=true
-                if [[ $checked_count -lt 3 ]]; then
-                  echo "  ✓ FOUND exact match: '$updated_pair'"
-                fi
-                break
+
+          if [[ "$RUN_FOR_ALL_NETWORKS" != "true" ]]; then
```

Comment:
Should we also verify and normalize when running for multiple networks?

Reply:
Thanks, good catch. I updated this (and also improved the pretty messy output when running single network).